Mathlib/RingTheory/Ideal/Maps.lean

```diff
@@ -385,6 +398,51 @@ theorem mem_map_of_equiv {E : Type*} [EquivLike E R S] [RingEquivClass E R S] (e
   · rintro ⟨x, hx, rfl⟩
     exact mem_map_of_mem e hx
 
+section Bijective
+
+/-- Special case of the correspondence theorem for isomorphic rings -/
+def relIsoOfBijective (hf : Function.Bijective f) : Ideal S ≃o Ideal R where
+  toFun := comap f
+  invFun := map f
+  left_inv := map_comap_of_surjective _ hf.2
+  right_inv J := le_antisymm (fun _ h ↦
+    have ⟨y, hy, eq⟩ := (mem_map_iff_of_surjective _ hf.2).mp h; hf.1 eq ▸ hy) le_comap_map
+  map_rel_iff' {_ _} := by
+    refine ⟨fun h ↦ ?_, comap_mono⟩
+    have := map_mono (f := f) h
+    simpa only [Equiv.coe_fn_mk, map_comap_of_surjective f hf.2] using this
+
+theorem comap_le_iff_le_map (hf : Function.Bijective f) {I : Ideal R} {K : Ideal S} :
+    comap f K ≤ I ↔ K ≤ map f I :=
+  ⟨fun h => le_map_of_comap_le_of_surjective f hf.right h, fun h =>
+    (relIsoOfBijective f hf).right_inv I ▸ comap_mono h⟩
+
+lemma comap_map_of_bijective (hf : Function.Bijective f) {I : Ideal R} :
+    (I.map f).comap f = I :=
+  le_antisymm ((comap_le_iff_le_map f hf).mpr fun _ ↦ id) le_comap_map
+
+theorem map.isMaximal (hf : Function.Bijective f) {I : Ideal R} (H : IsMaximal I) :
+    IsMaximal (map f I) := by
+  rw [isMaximal_def] at H ⊢
+  exact ((relIsoOfBijective _ hf).symm.isCoatom_iff _).mpr H
+
+theorem comap.isMaximal (hf : Function.Bijective f) {I : Ideal S} (H : IsMaximal I) :
+    IsMaximal (comap f I) := by
+  rw [isMaximal_def] at H ⊢
+  exact ((relIsoOfBijective _ hf).isCoatom_iff _).mpr H
+
+/-- A ring isomorphism sends a maximal ideal to a maximal ideal. -/
+instance map_isMaximal_of_equiv {E : Type*} [EquivLike E R S] [RingEquivClass E R S] (e : E)
+    {p : Ideal R} [hp : p.IsMaximal] : (map e p).IsMaximal :=
+  map.isMaximal e (EquivLike.bijective e) hp
+
+end Bijective
+
+theorem RingEquiv.bot_maximal_iff (e : R ≃+* S) :
```

Comment:
Should this not be for RingEquivClass?

Reply:
Then dot notation would not work. I merely moved this and didn't change it. If we want to change it I think assuming Bijective is the most general; we have to fix downstream uses but the same is true if we change it to RingEquivClass.

Reply:
Sorry, dot notation doesn't work from the start because name starts with `Ideal.RingEquiv` ...